analytics/analytics.go

```diff
@@ -0,0 +1,257 @@
+package analytics
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/dagger/dagger/core/pipeline"
+	"github.com/dagger/dagger/engine"
+	"github.com/vito/progrock"
+)
+
+const (
+	flushInterval = 1 * time.Second
+	trackURL      = "https://api.dagger.cloud/analytics"
+)
+
+type Event struct {
+	Timestamp  time.Time         `json:"ts,omitempty"`
+	Type       string            `json:"type,omitempty"`
+	Properties map[string]string `json:"properties,omitempty"`
+
+	DeviceID string `json:"device_id,omitempty"`
+	OrgID    string `json:"org_id,omitempty"`
+	IP       string `json:"ip,omitempty"`
```

Comment:
Looks like these are unused - should they be removed? (I was about to ask about GDPR related things; these two seem like the highest-risk.)

Reply:
Done!